components/datetime/src/datetime.rs

```diff
@@ -0,0 +1,259 @@
+    pub(super) symbols: Cow<'d, provider::gregory::DateSymbolsV1>,
+}
+
+impl<'d> DateTimeFormat<'d> {
+    /// `DateTimeFormat` constructor which takes a selected `Locale`, reference to a `DataProvider` and
+    /// a list of options and collects all data necessary to format date and time values into the given locale.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use icu_locid::Locale;
+    /// use icu_locid_macros::langid;
+    /// use icu_datetime::{DateTimeFormat, DateTimeFormatOptions};
+    /// use icu_datetime::mock::datetime::MockDateTime;
+    /// use icu_provider::inv::InvariantDataProvider;
+    ///
+    /// let locale: Locale = langid!("en").into();
+    ///
+    /// let provider = InvariantDataProvider;
+    ///
+    /// let options = DateTimeFormatOptions::default();
+    ///
+    /// let dtf = DateTimeFormat::try_new(locale, &provider, &options);
+    ///
+    /// assert_eq!(dtf.is_ok(), true);
+    /// ```
+    pub fn try_new<T: Into<Locale>, D: DataProvider<'d, provider::gregory::DatesV1> + ?Sized>(
+        locale: T,
+        data_provider: &D,
+        options: &DateTimeFormatOptions,
+    ) -> Result<Self, DateTimeFormatError> {
+        let locale = locale.into();
+        let data = data_provider
+            .load_payload(&DataRequest {
+                resource_path: ResourcePath {
+                    key: provider::key::GREGORY_V1,
+                    options: ResourceOptions {
+                        variant: None,
+                        langid: Some(locale.clone().into()),
+                    },
+                },
+            })?
+            .payload
+            .take()?;
+
+        let pattern = data
+            .patterns
+            .get_pattern_for_options(options)?
+            .unwrap_or_default();
+
+        let time_zone_field = pattern
+            .items()
+            .iter()
+            .filter_map(|p| match p {
+                PatternItem::Field(field) => Some(field),
+                _ => None,
+            })
+            .find(|field| matches!(field.symbol, FieldSymbol::TimeZone(_)));
+
+        if let Some(field) = time_zone_field {
+            return Err(DateTimeFormatError::UnsupportedField(field.symbol));
+        }
+
+        Ok(Self::new(locale, pattern, data))
+    }
+
+    /// Creates a new `DateTimeFormat` regardless of whether there are time-zone symbols in the pattern.
+    ///
+    /// By contrast, the public `DateTimeFormat::try_new` function will return an `Err` if there are
+    /// time-zone symbols in the pattern.
+    ///
+    /// This function is only `pub(super)` (not `pub`) because it is needed by `ZonedDateTimeFormat`
+    /// to create a `DateTimeFormat` for use internally. `ZonedDateTimeFormat` maintains
+    /// the invariant that `DateTimeFormat` will not be used to format the time zone.
+    ///
+    /// Creating a `DateTimeFormat` with time-zone symbols should always be an error
+    /// in public contexts.
+    pub(super) fn new<T: Into<Locale>>(
```

Comment:
`suggestion (optional)`: `new_unchecked`

Reply:
I'm torn. 

I like the idea of `new_unchecked` in theory based on what it communicates, but most `unchecked` Rust functions (in my experience) are unsafe. This function is safe.

Given that this is not a public-facing function, and won't be documented in the public API, I could go either way. 

Thoughts?

Reply:
Sounds good. Leaving it as just `new()` for now.